Scribe-i18n/scripts/iOS/convert_jsons_to_xcstrings.py

```diff
@@ -10,35 +10,53 @@
 import os
 
 directory = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
-json_dir_list = os.listdir(os.path.join(directory, "jsons"))
+# Define the path to the "jsons" folder
+jsons_folder = os.path.join(directory, "jsons")
+
+# Ensure the "jsons" folder exists inside the directory
+if not os.path.exists(jsons_folder):
+    print(f"Error: The folder '{jsons_folder}' does not exist. Please ensure the path is correct.")
+    exit(1)
+
+
+json_dir_list = os.listdir(jsons_folder)
 languages = sorted(
     [file.replace(".json", "") for file in json_dir_list if file.endswith(".json")]
 )
-path = os.path.join(os.path.join(directory, "jsons"), "en-US.json")
-file = open(path, "r").read()
-file = json.loads(file)
+
+# Load the base language file safely
+try:
+    with open(os.path.join(jsons_folder, "en-US.json"), "r") as json_file:
+        base_language_data = json.load(json_file)
+except FileNotFoundError:
+    print("Error: The base language file 'en-US.json' does not exist.")
+    exit(1)
+
 
 data = {"sourceLanguage": "en"}
 strings = {}
-for key in file:
-    language = {}
 
-    for lang in languages:
-        lang_json = json.loads(
-            open(os.path.join(os.path.join(directory, "jsons"), f"{lang}.json"), "r").read()
-        )
+# Pre-load all JSON files into a dictionary
+lang_data = {}
+for lang in languages:
+    with open(os.path.join(jsons_folder, f"{lang}.json"), "r") as lang_file:
+        lang_data[lang] = json.load(lang_file)
 
-        translation = lang_json[key] if key in lang_json else ""
+for key in base_language_data:          # Iterate over each key in the base language file
```

Comment:
These commends though don't need punctuation as you have them, but then as they're inline the first letter shouldn't be capitalized. The first word of a comment is only capitalized when it's its own line.

Reply:
> These commends though don't need punctuation as you have them, but then as they're inline the first letter shouldn't be capitalized. The first word of a comment is only capitalized when it's its own line.

I’ve made the edits to the comments as suggested. Thank you for pointing out the details; I’m learning a lot from your feedback!